mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/impl/ClientState.scala

```diff
@@ -0,0 +1,520 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.mqtt.streaming
+package impl
+
+import akka.NotUsed
+import akka.actor.typed.{ActorRef, Behavior, PostStop, Terminated}
+import akka.actor.typed.scaladsl.{ActorContext, Behaviors}
+import akka.annotation.InternalApi
+import akka.stream.{Materializer, OverflowStrategy}
+import akka.stream.scaladsl.{BroadcastHub, Keep, Source, SourceQueueWithComplete}
+import akka.util.Timeout
+
+import scala.concurrent.duration.FiniteDuration
+import scala.util.control.NoStackTrace
+import scala.util.{Failure, Success}
+
+/*
+ * A client connector is a Finite State Machine that manages MQTT client
+ * session state. A client connects to a server, subscribes/unsubscribes
+ * from topics to receive publications on and publishes to its own topics.
+ */
+@InternalApi private[streaming] object ClientConnector {
+
+  type ConnectData = Option[_]
+
+  /*
+   * A PINGREQ failed to receive a PINGRESP - the connection must close
+   *
+   * http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html
+   * 3.1.2.10 Keep Alive
+   */
+  case object PingFailed extends Exception with NoStackTrace
+
+  /*
+   * Construct with the starting state
+   */
+  def apply(consumerPacketRouter: ActorRef[RemotePacketRouter.Request[Consumer.Event]],
+            producerPacketRouter: ActorRef[LocalPacketRouter.Request[Producer.Event]],
+            subscriberPacketRouter: ActorRef[LocalPacketRouter.Request[Subscriber.Event]],
+            unsubscriberPacketRouter: ActorRef[LocalPacketRouter.Request[Unsubscriber.Event]],
+            settings: MqttSessionSettings)(implicit mat: Materializer): Behavior[Event] =
+    disconnected(
+      Uninitialized(consumerPacketRouter,
+                    producerPacketRouter,
+                    subscriberPacketRouter,
+                    unsubscriberPacketRouter,
+                    settings)
+    )
+
+  // Our FSM data, FSM events and commands emitted by the FSM
+
+  sealed abstract class Data(val consumerPacketRouter: ActorRef[RemotePacketRouter.Request[Consumer.Event]],
+                             val producerPacketRouter: ActorRef[LocalPacketRouter.Request[Producer.Event]],
+                             val subscriberPacketRouter: ActorRef[LocalPacketRouter.Request[Subscriber.Event]],
+                             val unsubscriberPacketRouter: ActorRef[LocalPacketRouter.Request[Unsubscriber.Event]],
+                             val settings: MqttSessionSettings)
+  final case class Uninitialized(
```

Comment:
Empty lines aren't that expensive, please add a few.
(Possibly putting some Scaladoc on the super-class to keep the grouping you tried to achieve.)

Reply:
I think that this is a coding style thing, and perhaps a little subjective. I won't mind at all if you'd like to do this once my PR is merged.